source/mod_dist.f90

```diff
@@ -98,159 +231,586 @@ subroutine dist_parseInputLine(inLine, iLine, iErr)
 
 end subroutine dist_parseInputLine
 
+! ================================================================================================ !
+!  Parse File Column Formats
+!  V.K. Berglyd Olsen, BE-ABP-HSS
+!  Created: 2019-07-05
+!  Updated: 2019-07-09
+! ================================================================================================ !
+subroutine dist_setColumnFormat(fmtName, fErr)
+
+  use crcoall
+  use string_tools
+  use numerical_constants
+
+  character(len=*), intent(in)  :: fmtName
+  logical,          intent(out) :: fErr
+
+  character(len=20) fmtBase
+  character(len=10) fmtUnit
+  real(kind=fPrec)  uFac
+  integer           c, unitPos, fmtLen
+
+  fErr = .false.
+
+  fmtLen = len_trim(fmtName)
+  if(fmtLen < 1) then
+    write(lerr,"(a)") "DIST> ERROR Unknown column format '"//trim(fmtName)//"'"
+    fErr = .true.
+  end if
+
+  unitPos = -1
+  do c=1,fmtLen
+    if(fmtName(c:c) == "[") then
+      unitPos = c
+      exit
+    end if
+  end do
+
+  if(unitPos > 1 .and. fmtLen > unitPos+1) then
+    fmtBase = trim(fmtName(1:unitPos-1))
+    fmtUnit = trim(fmtName(unitPos:fmtLen))
+  else
+    fmtBase = trim(fmtName)
+    fmtUnit = "[none]"
+  end if
+
+  select case(chr_toUpper(fmtBase))
+
+  case("OFF","SKIP") ! Ignored
+    call dist_appendFormat(dist_fmtNONE, one)
+  case("ID") ! Particle ID
+    call dist_appendFormat(dist_fmtPartID, one)
+  case("PARENT") ! Parent ID
+    call dist_appendFormat(dist_fmtParentID, one)
+
+  case("X") ! Horizontal position
+    call dist_unitScale(fmtName, fmtUnit, 1, uFac, fErr)
+    call dist_appendFormat(dist_fmtX, uFac)
+  case("Y") ! Vertical positiom
+    call dist_unitScale(fmtName, fmtUnit, 1, uFac, fErr)
+    call dist_appendFormat(dist_fmtY, uFac)
+  case("XP") ! Horizontal angle
+    call dist_unitScale(fmtName, fmtUnit, 2, uFac, fErr)
+    call dist_appendFormat(dist_fmtXP, uFac)
+  case("YP") ! Vertical angle
+    call dist_unitScale(fmtName, fmtUnit, 2, uFac, fErr)
+    call dist_appendFormat(dist_fmtYP, uFac)
+  case("PX") ! Horizontal momentum
+    call dist_unitScale(fmtName, fmtUnit, 3, uFac, fErr)
+    call dist_appendFormat(dist_fmtPX, uFac)
+  case("PY") ! Vertical momentum
+    call dist_unitScale(fmtName, fmtUnit, 3, uFac, fErr)
+    call dist_appendFormat(dist_fmtPY, uFac)
+  case("PX/P0","PXP0") ! Relative horizontal momentum
```

Comment:
I find "PXP0" is a bit confusing, but I see the need. Would it be possible to use P0 as unit, like PX[P0]?

Reply:
As discussed, we'll keep it this way.